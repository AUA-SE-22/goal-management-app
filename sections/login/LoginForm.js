// @mui
import { Stack, Button } from '@mui/material';
// hooks
import { signIn } from 'next-auth/react';
// components

export default function LoginForm() {
  return (
    <Stack spacing={3}>
      <Button fullWidth size="large" type="submit" variant="contained" onClick={() => signIn()}>
        Login
      </Button>
    </Stack>
  );
}
