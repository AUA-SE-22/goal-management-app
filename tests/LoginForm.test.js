import { render, screen } from '@testing-library/react'
import LoginForm from '../sections/login/LoginForm'

describe('Correctly renders', () => {

    it('the interface to login', () => {
    const { container } = render(<LoginForm />)
    const button = screen.getByRole('button', {
    name: 'Login',
    })

    expect(button).toBeInTheDocument()
    expect(container).toMatchSnapshot()
    })
})