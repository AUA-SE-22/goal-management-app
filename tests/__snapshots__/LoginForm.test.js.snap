// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders the interface to login 1`] = `
<div>
  <div
    class="css-1lwbda4-MuiStack-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-189e9pj-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Login
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
